tests/unit/mainwindow/test_prompt.py

```diff
@@ -93,6 +95,47 @@ def test_backspacing_path(self, qtbot, tmp_path, get_prompt):
         prompt.item_focus('next')
         prompt.item_focus('next')
         assert prompt._lineedit.text() == str(testdir / 'bar')
+    
+    def test_filtering_path(self, qtbot, tmp_path, get_prompt):
```

Comment:
The test looks good from a quick look, but please make sure to use [pytest parametrization](https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-mark-parametrize) to reduce the code duplication and turn this into three independent tests. You could for example parametrize a list of keys to press, and the output you expect with that input.